vault/identity_store_aliases.go

```diff
@@ -138,10 +140,7 @@ func (i *IdentityStore) handleAliasCreateUpdate() framework.OperationFunc {
 		customMetadata := make(map[string]string)
 		data, customMetadataExists := d.GetOk("custom_metadata")
 		if customMetadataExists {
-			err = mapstructure.Decode(data, &customMetadata)
-			if err != nil {
-				return nil, err
-			}
+			customMetadata = data.(map[string]string)
```

Comment:
Could a client craft a request that causes a panic if they set `custom_metadata` to something that's not a `map[string]string`?

Reply:
This is not my PR, but I reviewed the original prior to the backport. This type assertion should be safe due to the `GetOk` call above. The `custom_metadata` field is defined as `framework.TypeKVPairs` which is ultimately [parsed](https://github.com/hashicorp/vault/blob/7e1c19128ceea80df271ac0e4c0ffa76dfa1b1cd/sdk/framework/field_data.go#L295-L316) as `map[string]string`.

Reply:
Thanks for the explainer! TIL